Fixed & Added scripts

kmsiapps opened the pull request:
* Fixed broken spawn_protect.py (Fixed typo)

* Added Spade-nade bug block script(spadenadefix.py)
(If you held spade and change tool into grenades with RMB clicked, the grenade would throw and explode instantly. This script fixes it)

noway421 commented on feature_server/config/scripts/spadenadefix.py:
```diff
@@ -0,0 +1,17 @@
+"""
+def apply_script(protocol,connection,config):
+
+    class kmsiConnection(connection):
```
Would be better to use established naming convention in the form of "${scriptname}Connection"

Change it to `SpadenadeFixConnection` in that case?

noway421 commented on feature_server/config/scripts/spawn_protect.py:
```diff
@@ -2,30 +2,28 @@
 def apply_script(protocol, connection, config):
     spawn_protect_time = config.get('spawn_protect_time', 3.0)
-
+    
```
Please don't leave trailing white space

noway421 commented on feature_server/config/scripts/spawn_protect.py:
```diff
@@ -2,30 +2,28 @@
-
-    return protocol, SpawnProtectConnection
+            return connection.on_hit(self, hit_amount, hit_player, type, grenade)
+    
```
Please remove trailing white space

feikname commented on feature_server/config/scripts/spawn_protect.py:
```diff
@@ -2,30 +2,28 @@
-    return protocol, SpawnProtectConnection
+            return connection.on_hit(self, hit_amount, hit_player, type, grenade)
+    
+    return protocol, SpawnProtectConnection
```
The final line doesn't end with a EOL here.

kmsiapps replied:
@feikname Mmm.. I don't know what are you meaning. Can you explain it more easily?

noway421 replied:
An empty line as the last line of the file. Not all text editors insert it automatically, although they should. Right now it ends abrubtly.

Basically you know you got it right when you get rid of this little icon in github file view: 
<img width="409" alt="screen shot 2017-01-22 at 8 04 34 pm" src="https://cloud.githubusercontent.com/assets/2031472/22180752/4ecd39a8-e0de-11e6-9a65-d6fd8ed0a85b.png">

kmsiapps replied:
@noway421 Aha, I got it. thanks for the kind explanation!

feikname replied:
@kmsiapps [this StackOverflow question](http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) explains it quite well.

Basically, some editors won't place the End-of-line (*[EOL](https://en.wikipedia.org/wiki/Newline)*) character at the end of the last line unless you add an additional one below. However, that EOL character is necessary for some programs to work correctly with the file. 

noway421 tip about getting rid of the little icon is pretty good too :)

kmsiapps replied:
@feikname I updated my scripts again. Is it okay now?

feikname commented on feature_server/config/scripts/spadenadefix.py:
```diff
@@ -0,0 +1,17 @@
+by kmsi(kmsiapps@gmail.com)
+version 1(2017.01.21)
+"""
+def apply_script(protocol,connection,config):
```
Could you please add spaces after the commas? To improve readability :)

kmsiapps commented:
* Removed trailing spaces, changed protocol name into Spadenadeconnection
* Added nosharp.py - It would hopefully fix this problem : https://github.com/piqueserver/piqueserver/issues/3

Needed some test, although I made some test before commit.

feikname commented:
It's very common to have a bunch of commits :P

noway421 commented:
Looks good to me :+1:
